components/authz-service/projectassignment/errors.go

```diff
@@ -0,0 +1,45 @@
+package projectassignment
+
+import (
+	"fmt"
+)
+
+// ProjectsMissingErr occurs when some of the projects in the project diff
+// did not exist.
+type ProjectsMissingErr struct {
+	projectsMissing []string
+}
+
+func NewProjectsMissingError(projectsMissing []string) error {
+	return &ProjectsMissingErr{projectsMissing: projectsMissing}
+}
+
+func (e *ProjectsMissingErr) Error() string {
+	var errorStr string
+	if len(e.projectsMissing) > 1 {
+		errorStr = fmt.Sprintf("You attempted to modify projects for this object but some did not exist: %q", e.projectsMissing)
```

Comment:
Three things:
1. I think it is more clear to use present tense instead of past tense in all these error messages.
2. I think saying "these" instead of "some" would help clarity, too: when I ran a curl I was not sure whether it was saying (a) just some--but not all--of the ones listed here do not exist  or (b) these are, in fact, all the ones that do not exist.
3. Not sure that we've used the more verbose, conversational style of this error message before (i.e. compared to just saying "These projects do not exist: %q"). I have no preference either way, just raising the question.

Reply:
Updating to be a bit more clear.